sat/cli/bmccreds/creds_manager.py

```diff
@@ -67,12 +67,18 @@ class BMCCredsManager:
     CHASSIS_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+b\d+$')
     ROUTER_BMC_XNAME_REGEX = re.compile(r'x\d+c\d+r\d+b\d+$')
 
+    # TODO: Allow passing in either an explicit password or a PasswordGenerator instance
+    # TODO: Define the PasswordGenerator class to generate passwords
     def __init__(self, password, xnames, domain, force, report_format):
         self.password = password
         self.xnames = [XName(xname) for xname in xnames]
         self.domain = domain
         self.force = force
         self.report_format = report_format
+        self.password_info = {
+            'length': len(password) if password else RANDOM_PASSWORD_LENGTH,
+            'allowed_chars': password or VALID_CHAR_SETS
```

Comment:
Oh, I missed this in the initial review comments, but we want to support passing in the random password length and the character set. If you add these as arguments to the `__init__` method, that would work.

Reply:
Addressed